framework/include/utils/FVUtils.h

```diff
@@ -300,6 +352,60 @@ determineElemOneAndTwo(const FaceInfo & fi, const MooseVariableFV<OutputType> &
 
   return std::make_tuple(elem_one, elem_two, one_is_elem);
 }
+
+/**
+ * From Moukalled 12.30
+ *
+ * r_f = (phiC - phiU) / (phiD - phiC)
+ *
+ * However, this formula is only clear on grids where upwind locations can be readily determined,
+ * which is not the case for unstructured meshes. So we leverage a virtual upwind location and
+ * Moukalled 12.65
+ *
+ * phiD - phiU = 2 * grad(phi)_C * dCD ->
+ * phiU = phiD - 2 * grad(phi)_C * dCD
+ *
+ * Combining the two equations and performing some algebraic manipulation yields this equation for
+ * r_f:
+ *
+ * r_f = 2 * grad(phi)_C * dCD / (phiD - phiC) - 1
+ *
+ * This equation is clearly asymmetric considering the face between C and D because of the
+ * subscript on grad(phi). Hence this method can be thought of as constructing an r associated with
+ * the C side of the face
+ */
+template <typename Scalar, typename Vector>
+ADReal
+rF(const Scalar & phiC, const Scalar & phiD, const Vector & gradC, const RealVectorValue & dCD)
+{
+  if ((phiD - phiC) == 0)
+    return 1e6 * MathUtils::sign(gradC * dCD) + 0 * (gradC * dCD + phiD + phiC);
+
+  return 2. * gradC * dCD / (phiD - phiC) - 1.;
 }
 
+/**
+ * Interpolates with a limiter
+ */
+template <typename Scalar, typename Vector>
+ADReal
+interpolate(const Limiter & limiter,
+            const Scalar & phiC,
+            const Scalar & phiD,
+            const Vector & gradC,
+            const FaceInfo & fi,
+            const bool C_is_elem)
+{
+  // Using beta, w_f, g nomenclature from Greenshields
+  const auto r_f = rF(phiC, phiD, gradC, C_is_elem ? fi.dCF() : RealVectorValue(-fi.dCF()));
+  const auto beta = limiter(r_f);
+
+  const auto w_f = C_is_elem ? fi.gC() : (1. - fi.gC());
```

Comment:
Hmm you are right. Obviously these reduce to the same thing for non-skewed meshes, but...

I like the way that we have defined `gC` and consequently `w_f` because if were to sum the `gC` taken from the element perspective, and then from the neighbor perspective, then we would get 1. But perhaps I should document our different definition of `w_f`...?